CMakeLists.txt

```diff
@@ -92,7 +92,9 @@ else ()
             message(FATAL_ERROR "Core Foundation not found")
         endif ()
 
-        list(APPEND PLATFORM_LIBS dl Threads::Threads ${CORE_FOUNDATION_LIB})
+        list(APPEND PLATFORM_LIBS dl Threads::Threads)
+        # Don't add the exact path to CoreFoundation as this would hardcode the SDK version
+        list(APPEND PLATFORM_LINK_FLAGS -framework CoreFoundation)
```

Comment:
If you want to avoid the extra variable, according to [this SO answer](https://stackoverflow.com/a/28313304) you should be able to add `"-framework CoreFoundation"` to `PLATFORM_LIBS`